examples/reduction/reduction/reduction.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from enum import IntEnum
+from typing import Any
+
+import cunumeric as num
+import pyarrow as pa
+
+import legate.core.types as ty
+from legate.core import Array, Store
+
+from .library import user_context as context, user_lib
+
+
+class OpCode(IntEnum):
+    BINCOUNT = user_lib.cffi.BINCOUNT
+    CATEGORIZE = user_lib.cffi.CATEGORIZE
+    HISTOGRAM = user_lib.cffi.HISTOGRAM
+    MATMUL = user_lib.cffi.MATMUL
+    MUL = user_lib.cffi.MUL
+    SUM_OVER_AXIS = user_lib.cffi.SUM_OVER_AXIS
+
+
+class _Wrapper:
+    def __init__(self, store: Store) -> None:
+        self._store = store
+
+    @property
+    def __legate_data_interface__(self) -> dict[str, Any]:
```

Comment:
Should this just be a part of the Store? Shouldn't we just have `Store` define its own `__legate_data_interface__` ?

Reply:
Yes, that seems like a better approach. I assume that the implementation here serves both as a stop-gap measure and also guides and motivates the implementation of such needed features.